### PR TITLE
gen2 profiles (design proposal 3)

### PR DESCRIPTION
**Pull request by bcle**
Design doc: https://github.com/arlonproj/arlon/blob/private/leb/multiprofiles/docs/gen2_profiles_proposal_3.md
Testing:
-  unit test located at pkg/appprofile/appprofile_test.go
- E2E test: make e2e-appprofiles

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,16 @@ else
...
 
+## Location to install dependencies to
+LOCALBIN ?= $(shell pwd)/bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+
+## Tool Versions
+CONTROLLER_TOOLS_VERSION ?= v0.9.2
```
Comment (cruizen) [inline Makefile:31]: We have been using version v0.10.0 in the main branch (and using it to generate the manifests) since a few weeks.

Comment (bcle) [reply Makefile]: Fixed.

**Comment by cruizen** [comment]
Fixes #98

File: pkg/app/app.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/arlonproj/arlon/pkg/ctrlruntimeclient"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	restclient "k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const ProfileLabelKey = "arlon.io/profile"
```
Comment (Rohitrajak1807) [inline pkg/app/app.go:17]: seems to be unused

File: pkg/cluster/external.go [hunk 3]
```diff
@@ -3,21 +3,20 @@ package cluster
 import (
```
Comment (Rohitrajak1807) [inline pkg/cluster/external.go:3]: If I understand it correctly, this file has the logic to manage and unmanage external clusters using the new profiles, is it?

**Review by Rohitrajak1807 — APPROVED** [review]
Added some general comments and nitpicks. I'm approving the changes, because the suggestions in my review are mostly related to cleanup which can be done as a part of the rebase.

**Review by cruizen — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -75,6 +85,9 @@ clean:
 build: generate fmt vet cluster-config ## Build manager binary.
 	go build -ldflags '$(LD_FLAGS)' -o bin/arlon main.go
 
+unittests:
```
[the remark this answers is not in the document]
Comment (bcle) [reply Makefile:88]: removed target

File: pkg/cluster/external.go (hunk 3 above)
Comment (bcle) [reply pkg/cluster/external.go:3]: No, it uses old profiles.

**Pull request merged**